README.pod

```diff
@@ -49,6 +49,8 @@ plugins provided by the distribution are these:
 
 =item * L<Git::Hooks::CheckCommit> - enforce commit policies
 
+=item * L<Git::Hooks::CheckDiff> - check differences between commits
```

Comment:
Yes. The idea is that this hook will generate a git-diff and pass its output to one or more commands to check it. It can be used in a pre-commit or in a pre-receive hook, for example. In a pre-commit it uses git-diff-index to generate the next commit diff. In a pre-receive it uses a git-diff-tree to generate the next push diff.

How could I make this clearer in a single line?

Reply:
I think the message is good enough, don't worry about it. I was a little confused because the spaces verification happens in the whole file and not just in the changed/added lines, I thought this new feature would work the same way. Actually, it would be good if the user could choose if he/she wants to verify the whole file or just the changed/added lines. What do you think?

Reply:
There is the Git::Hooks::CheckFile hook for checking the whole files. CheckDiff is meant to be more efficient and more specific when the check need only see the changed lines.

Reply:
Uhm... but we can't use "find-secret-leakage-in-git-diff.pl" in the CheckFile, can we?

Reply:
Nope. It's meant to check a diff. The idea is that you use it to check only newly added parts in your code to avoid introducing new leakages. I don't think CheckFile would be a good fit for finding secret leakages. If you want to check all of your code I think you should process all of your code at once, using something like SonarQube.

Anyway, this is such a small script that it would be easy to create another one called find-secret-leakage.pl that would check all the lines in the input, not bothering to parse a diff output. I wouldn't use it in a CheckFile, but it could be useful to check all of your files in a script.

Reply:
Ok